image-service/test.sh.in

```diff
@@ -0,0 +1,455 @@
+          vagrant destroy -f
+          #suspend all running instance
+          vagrantSuspendAll
+          #rmeove all VMs before retying
+          rm -rf "$HOME/VirtualBox VMs"
+          rm "$HOME/.config/VirtualBox/VirtualBox.xml"
+      else
+          echo "Vagrant up passed."
+          break
+      fi
+  done
+  if [ $result -ne 0 ]; then
+      echo "Vagrant up failed."
+      exit 1
+  fi
+}
+
+vagrantDestroy() {
+  cd ${WORKSPACE}/RackHD/example
+  vagrant destroy -f
+}
+
+vagrantHalt() {
+  cd ${WORKSPACE}/RackHD/example
+  vagrant halt
+}
+
+virtualBoxDestroyAll() {
+  set +e
+  for uuid in `vboxmanage list vms | awk '{print $2}' | tr -d '{}'`; do
+    echo "shutting down vm ${uuid}"
+    vboxmanage controlvm ${uuid} poweroff
+    echo "deleting vm ${uuid}"
+    vboxmanage unregistervm ${uuid}
+  done
+  set -e
+}
+
+vnc_record_start(){
+  pushd ${WORKSPACE}/RackHD/example # to use the Vagrantfile in it
+  export fname_prefix="vNode"
+  if [ ! -z $BUILD_ID ]; then
+      export fname_prefix=${fname_prefix}_b${BUILD_ID}
+  fi
+
+  #http://stackoverflow.com/questions/25331758/vagrant-ssh-c-and-keeping-a-background-process-running-after-connection-closed/27327955
+  #trying various ways to  detach the process from the shell,  but all not working 
+  # including vagrant ssh -c "nohup bash xx.sh & "
+  # including vagrant ssh -c "bash xx.sh " &
+  nohup vagrant ssh -c "cd /home/vagrant/src/build-config/;   bash vnc_record.sh /home/vagrant/log/ $fname_prefix" &
+  #/home/vagrant/log/ will be mapped to host(outside vagrant)'s $WORKSPACE/build-log
+}
+
+vnc_record_stop(){
+  #sleep 2 sec to ensure FLV finishes the disk I/O before VM destroyed
+  vagrant ssh -c 'set +e ;    pkill flvrec.py;  sleep 2 ' 
+
+}
+
+
+
+generateSolLog(){
+  cd ${WORKSPACE}/RackHD/example
+  vagrant ssh -c 'cd /home/vagrant/src/build-config/; \
+        bash generate-sol-log.sh' > ${WORKSPACE}/sol.log &
+}
+
+generateSysLog(){
+  cd ${WORKSPACE}/RackHD/example
+  vagrant ssh -c 'dmesg > /home/vagrant/log/dmesg.log'
+  vagrant ssh -c 'cp /var/log/syslog /home/vagrant/log/syslog.log'
+}
+
+generateMongoLog(){
+  cd ${WORKSPACE}/RackHD/example
+  vagrant ssh -c 'cp -r /var/log/mongodb /home/vagrant/log'
+}
+
+setupVirtualEnv(){
+  pushd ${WORKSPACE}/RackHD/test
+  rm -rf .venv/on-build-config
+  ./mkenv.sh on-build-config
+  source myenv_on-build-config
+  popd
+  if [ "$MODIFY_API_PACKAGE" == true ] ; then
+      apiPackageModify
+  fi
+}
+
+BASE_REPO_URL="${BASE_REPO_URL}"
+runTests() {
+  set +e
+  if [ "$RUN_FIT_TEST" == true ] ; then
+     cd ${WORKSPACE}/RackHD/test
+     #TODO Parameterize FIT args
+     python run_tests.py -test deploy/rackhd_stack_init.py -stack vagrant  -port 9090 -xunit
+     if [ $? -ne 0 ]; then
+         echo "Test FIT failed running deploy/rackhd_stack_init.py"
+         exit 1
+     fi
+     python run_tests.py -test tests -group imageservice -stack vagrant -extra imageservice_config.json -port 9090 -v 9 -xunit
```

Comment:
there will be 4 branches in PR Gate Func Test.
and for all those 4 branches, they will run the same imageservice test case .
and no OS-install being tested, is it be design ?